dev-docs/batch-system.rst

```diff
@@ -0,0 +1,784 @@
+   for the batch or an individual job to complete. The implementation
+   of the wait operation is by continuously polling the Batch Front
+   End until the batch state is "complete".
+
+
+~~~~~~~~~~
+Data Model
+~~~~~~~~~~
+
+The core concepts in the Batch data model are billing projects,
+batches, jobs, updates, attempts, and resources.
+
+A **billing project** is a mechanism for imposing cost control and
+enabling the ability to share information about batches and jobs
+across users. Each billing project has a list of authorized users and
+a billing limit. Any users in the billing project can view information
+about batches created in that billing project. Developers can
+add/delete users in a billing project and modify billing limits. Right
+now, these operations are manually done after a Batch user submits a
+formal request to the Hail team. Note that the Hail billing project is
+different than a GCP billing project.
+
+A **batch** is a set of **jobs**. Each batch is associated with a
+single billing project. A batch also consists of a set of
+**updates**. Each update contains a distinct set of jobs. Updates are
+distinct submissions of jobs to an existing batch in the system. They
+are used as a way to add jobs to a batch. A batch is always created
+with 0 updates and 0 total jobs. To add jobs to a batch, an update
+must be created with an additional API call and the number of jobs in
+the update must be known at the time of the API call. The reason for
+this is because an update reserves a block of job IDs in order to
+allow multiple updates to a batch to be submitted simultaneously
+without the need for locking as well as for jobs within the update to
+be able to reference each other before the actual job IDs are
+known. Once all of the jobs for a given batch update have been
+submitted, the update must be committed in order for the jobs to be
+visible in the UI and processed by the batch driver.
+
+A job can have **attempts**. An attempt is an individual execution
+attempt of a job running on a worker VM. There can be multiple
+attempts if a job is preempted. If a job is cancelled before it has a
+chance to run, it will have zero attempts. An attempt has the
+**instance** name that it ran on, the start time, and the end
+time. The end_time must always be greater than the start_time. All
+billing tracking is done at the level of an attempt as different
+attempts for the same job can have different resource pricing if the
+VM configurations are different (4 core worker vs 16 core worker).
+
+Billing is tracked by **resources**. A resource is a product (example:
+preemptible n1-standard-16 VM in us-central1) combined with a version
+tag. Each resource has a rate that is used to compute cost when
+multiplied by the usage of the resource. Resource rates are in units
+that are dependent on the type of resource. For example, VM rates are
+measured in mCPU*hours. Each attempt has a set of resources associated
```

Comment:
Google's rates are posted in core-hours, ours are in msec-mcpu, which are you referring to here?
```suggestion
denominated in USD per core-hour. Each attempt has a set of resources associated
```